### ISSUE TITLE
Doesn't report Tab character(s) in multiline string

### ISSUE THREAD

**Issue by schalkms** [pinterest/ktlint#575]
Please take a look at the following multiline string example.
The tabs in the multiline string don't get reported by ktlint 0.34.2.

```kotlin
package ktlint.sample

public const val str = """
	Tab at the beginning of this line
Tab	 in the middle of this string
Tab at the end of this line.	
"""
```

**Comment by paul-dingemans** [comment]
I don't think that the initial example given should report an error:
```
public const val str = """
	Tab at the beginning of this line
Tab	 in the middle of this string
Tab at the end of this line.	
"""
```
This multiline string is not suffixed with `.trimIndent()` nor `.trimMargin()`. So the entire content of the line should be preserved.
Note: the indentation rule of ktlint currently ignores the `.trimMargin()` suffix entirely.

Also no error should be reported in case the tab character is found on the right hand side of the margin like in example below:
```
    @Test
    fun `format multiline string with tabs after the margin is indented properly`() {
        val code =
            """
            val str = ${'"'}${'"'}${'"'}
                 ${'\t'}Tab at the beginning of this line but after the right hand side of the indentation margin
                 Tab${'\t'}in the middle of this string
                 Tab at the end of this line.${'\t'}
                 ${'"'}${'"'}${'"'}.trimIndent()
                 """.trimIndent()
        assertThat(IndentationRule().lint(code)).isEmpty()
    }
```

Only in case the tab character is found on the left hand side of the margin the error should be reported:
```
    @Test
    fun `format multiline string with tabs before the margin is indented properly`() {
        val code =
            """
            val str = ${'"'}${'"'}${'"'} 
${'\t'}       Tab at the beginning of this line but on the left hand side of the indentation margin
                 Tab${'\t'}in the middle of this string
                 Tab at the end of this line.${'\t'}
                 ${'"'}${'"'}${'"'}.trimIndent()
                 """.trimIndent()
        assertThat(IndentationRule().lint(code)).isEqualTo(
            listOf(
                LintError(2, 11, "indent", "Unexpected tab character(s) in indent of multiline string"),
            )
        )
    }
```

Can you agree with above? I have not yet prepared the PR because the calculation of the position of the unexpected tab needs to be determined. Before I do that, I would like to have confirmation about this approach.

**Comment by schalkms** [comment]
Thanks for the clarification. I understand the reasons behind this.
For me it is fine if the maintainers (from Pinterest) declare this as system behavior.
Since this issue exists for quite some time, who can help with the decision here to get this resolved? - @romtsn 

> Note: the indentation rule of ktlint currently ignores the .trimMargin() suffix entirely.

I didn't know that. Thanks for the hint.

**Comment by romtsn** [comment]
Yeah, I agree that we shouldn't touch the contents of raw strings if they are not using `trimIndent` as it might be that they are using tabs intentionally (and it's hard for us to distinguish between intentional/unintentional usage).

But it would be beneficial to detect those in the beginning of the line and where `trimIndent` is present, so @paul-dingemans your test is describing it well (with a caveat that this shouldn't be a violation in case `indent_style=tab`)

**Comment by paul-dingemans** [comment]
Ok, I am already working on a fix. It will be ready early 2021.

**Comment by paul-dingemans** [comment]
As discussed in above PR, the indent rule will never change the indentation inside a raw string literal. Although it can be very useful to fix indentation inside string literals (appended with trimIndent), it is expected that such behaviour is not well expected in the entire community. Therefore this behaviour will not be added to the indent rule which is contained inside the standard ruleset.
@romtsn Will you close this issue, as it was your decision not to resolve it?

I am working on a publicly available custom ruleset in which such a rule will be provided. Either this rule will be provided via a separate repository or it might be included in the experimental ruleset of ktlint. For more info see issue https://github.com/pinterest/ktlint/issues/1229 and PR https://github.com/pinterest/ktlint/pull/1230.

**Comment by romtsn** [comment]
Yeah, @schalkms sorry, but we aren't going to support indentation within string templates for now
